packages/gatsby/src/commands/develop.ts

```diff
@@ -1,613 +1,200 @@
-                res.writeHead(response.statusCode || 200, response.headers)
-              )
-              .on(`error`, (err, _, response) => {
-                if (response) {
-                  res.writeHead(response.statusCode || 400, response.headers)
-                } else {
-                  const message = `Error when trying to proxy request "${req.originalUrl}" to "${proxiedUrl}"`
-
-                  report.error(message, err)
-                  res.sendStatus(500)
-                }
-              })
-          )
-          .pipe(res)
+    return new Promise(resolve => {
+      this.process.on(`exit`, () => {
+        this.process.removeAllListeners()
+        resolve()
       })
     })
   }
-
-  await apiRunnerNode(`onCreateDevServer`, { app })
-
-  // In case nothing before handled hot-update - send 404.
-  // This fixes "Unexpected token < in JSON at position 0" runtime
-  // errors after restarting development server and
-  // cause automatic hard refresh in the browser.
-  app.use(/.*\.hot-update\.json$/i, (_, res) => {
-    res.status(404).end()
-  })
-
-  // Render an HTML page and serve it.
-  app.use((_, res) => {
-    res.sendFile(directoryPath(`public/index.html`), err => {
-      if (err) {
-        res.status(500).end()
-      }
-    })
-  })
-
-  /**
-   * Set up the HTTP server and socket.io.
-   * If a SSL cert exists in program, use it with `createServer`.
-   **/
-  const server = program.ssl
-    ? https.createServer(program.ssl, app)
-    : new http.Server(app)
-
-  const socket = websocketManager.init({ server, directory: program.directory })
-
-  const listener = server.listen(program.port, program.host)
-
-  // Register watcher that rebuilds index.html every time html.js changes.
-  const watchGlobs = [`src/html.js`, `plugins/**/gatsby-ssr.js`].map(path =>
-    slash(directoryPath(path))
-  )
-
-  chokidar.watch(watchGlobs).on(`change`, async () => {
-    await createIndexHtml(indexHTMLActivity)
-    socket.to(`clients`).emit(`reload`)
-  })
-
-  return { compiler, listener, webpackActivity }
+  on(type, callback): void {
+    this.process.on(type, callback)
+  }
 }
 
 module.exports = async (program: IProgram): Promise<void> => {
-  // We want to prompt the feedback request when users quit develop
-  // assuming they pass the heuristic check to know they are a user
-  // we want to request feedback from, and we're not annoying them.
-  process.on(
-    `SIGINT`,
-    async (): Promise<void> => {
-      if (await userPassesFeedbackRequestHeuristic()) {
-        showFeedbackRequest()
-      }
-      process.exit(0)
-    }
+  const developProcessPath = resolveCwd.silent(
+    `gatsby/dist/commands/develop-process`
   )
+  // Run the actual develop server on a random port, and the proxy on the program port
+  // which users will access
+  const proxyPort = program.port
+  const developPort = await getRandomPort()
+
+  const script = new ControllableScript(`
+    const cmd = require("${developProcessPath}");
```

Comment:
Any reasoning behind needing to write this to disk to run it?

Reply:
Is there a better way to spawn a child process that runs an arbitrary Node.js script? 🤔

Reply:
We still write this but write it in `.cache` now